packages/flutter_tools/test/general.shard/ios/ios_device_start_nonprebuilt_test.dart

```diff
@@ -0,0 +1,296 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:file/memory.dart';
+import 'package:flutter_tools/src/application_package.dart';
+import 'package:flutter_tools/src/artifacts.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/build_info.dart';
+import 'package:flutter_tools/src/cache.dart';
+import 'package:flutter_tools/src/device.dart';
+import 'package:flutter_tools/src/ios/devices.dart';
+import 'package:flutter_tools/src/ios/ios_deploy.dart';
+import 'package:flutter_tools/src/ios/mac.dart';
+import 'package:flutter_tools/src/project.dart';
+import 'package:mockito/mockito.dart';
+import 'package:platform/platform.dart';
+import 'package:quiver/testing/async.dart';
+
+import '../../src/common.dart';
+import '../../src/context.dart';
+import '../../src/fake_process_manager.dart';
+
+const List<String> kRunReleaseArgs = <String>[
+  '/usr/bin/env',
+  'xcrun',
+  'xcodebuild',
+  '-configuration',
+  'Release',
+  '-quiet',
+  '-workspace',
+  'Runner.xcworkspace',
+  '-scheme',
+  'Runner',
+  'BUILD_DIR=/build/ios',
+  '-sdk',
+  'iphoneos',
+  'ONLY_ACTIVE_ARCH=YES',
+  'ARCHS=arm64',
+  'FLUTTER_SUPPRESS_ANALYTICS=true',
+  'COMPILER_INDEX_STORE_ENABLE=NO',
+];
+
+const String kConcurrentBuildErrorMessage = '''
+"/Developer/Xcode/DerivedData/foo/XCBuildData/build.db":
+database is locked
+Possibly there are two concurrent builds running in the same filesystem location.
+''';
+
+void main() {
```

Comment:
I removed most of the test functionality by calling startApp instead of going through the fake command runner